pkg/service/service.go

```diff
@@ -15,108 +15,172 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"os"
 	"time"
 
 	"github.com/edgexfoundry/device-sdk-go/internal/autoevent"
+	"github.com/edgexfoundry/device-sdk-go/internal/clients"
 	"github.com/edgexfoundry/device-sdk-go/internal/common"
 	"github.com/edgexfoundry/device-sdk-go/internal/container"
 	"github.com/edgexfoundry/device-sdk-go/internal/controller"
 	dsModels "github.com/edgexfoundry/device-sdk-go/pkg/models"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
 	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-registry/registry"
+	"github.com/gorilla/mux"
 
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
 	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
 	"github.com/google/uuid"
 )
 
 var (
-	svc *Service
+	sdk *DeviceServiceSDK
 )
 
-// A Service listens for requests and routes them to the right command
-type Service struct {
-	svcInfo     *common.ServiceInfo
-	asyncCh     chan *dsModels.AsyncValues
-	deviceCh    chan []dsModels.DiscoveredDevice
-	startTime   time.Time
-	controller  controller.RestController
-	initiazlied bool
+type DeviceServiceSDK struct {
```

Comment:
While it's great that you're trying to align with the app-functions-sdk-go on this naming, I really don't like this name. **SDK** is an acronym that means "software development kit", whereas what we're providing is a golang package which implements the core logic of an EdgeX device service. I honestly don't see the reason we even have to re-name `Service`, however if you feel it's too generic, then I'm not opposed to `DeviceService`, I'm just not a fan of including **SDK** in the name as it doesn't really make sense.

Reply:
@tonyespy , I disagree. To me `DeviceService` or `Service` is the thing being built using this SDK.  Obvious I am ok with `SDK` in the name... ;-)

Reply:
Actually, after think about this last night, I agree with @tonyespy. The SDK is more than this struct/functionality and this  `object` that is responsible for spinning up the server. Ideally we would have NewDeviceService() constructor function that the developers use.

Reply:
updated